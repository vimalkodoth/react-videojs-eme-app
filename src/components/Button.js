import React, { Component } from "react";
import { css } from "@emotion/core";
import arrow from "./../static/img/arrow.svg";
import { bool, func } from "prop-types";

/**
 * Button Component
 */
class Button extends Component {
    static propTypes = {
        onClick: func,
        show: bool,
        prev: bool
    };
    static defaultProps = {
        onClick: (val) => val,
        show: true,
        prev: true
    };
    render() {
        const { onClick, show, prev } = this.props;
        return (
            <div
                data-testid="button"
                css={ButtonStyles({ prev })}
                style={{
                    display: show ? "block" : "none"
                }}
                onClick={onClick}
            ></div>
        );
    }
}

export default Button;

const ButtonStyles = (props) => css`
    position: absolute;
    cursor: pointer;
    top: 50%;
    margin-top: -16px;
    width: 32px;
    height: 32px;
    @media (min-width: 560px) {
        margin-top: -24px;
        width: 48px;
        height: 48px;
    }
    background-image: url(${arrow});
    background-color: rgba(255, 255, 0, 0.75);
    background-size: 110%;
    background-repeat: no-repeat;
    background-position: -3px -2px;
    border-radius: 50%;
    z-index: 9999;
    ${props && props.prev
        ? `
    left: 5px;
  `
        : `
    right: 5px;
    transform: rotate(180deg);
  `}
`;
