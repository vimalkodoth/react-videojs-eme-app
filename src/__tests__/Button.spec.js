import React from "react";
import { shallow } from "enzyme";
import Button from "./../components/Button";

describe("Button", () => {
    it("renders correctly", () => {
        const onClick = jest.fn(),
            show = true,
            prev = true;
        const component = shallow(
            <Button onClick={onClick} show={show} prev={prev} />
        );
        expect(component).toMatchSnapshot();
    });
    it("invokes callback on click", () => {
        const onClick = jest.fn(),
            show = true,
            prev = true;
        const component = shallow(
            <Button onClick={onClick} show={show} prev={prev} />
        );
        component.simulate("click");
        expect(component.find('[data-testid="button"]').length).toEqual(1);
        expect(onClick).toHaveBeenCalledTimes(1);
    });
});
