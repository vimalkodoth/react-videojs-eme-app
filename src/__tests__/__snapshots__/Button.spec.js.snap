// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<EmotionCssPropInternal
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
  css={
    Object {
      "map": undefined,
      "name": "7x77wu-ButtonStyles",
      "next": undefined,
      "styles": "position:absolute;cursor:pointer;top:50%;margin-top:-16px;width:32px;height:32px;@media (min-width:560px){margin-top:-24px;width:48px;height:48px;}background-image:url();background-color:rgba(255,255,0,0.75);background-size:110%;background-repeat:no-repeat;background-position:-3px -2px;border-radius:50%;z-index:9999;
    left: 5px;
  ;label:ButtonStyles;",
      "toString": [Function],
    }
  }
  data-testid="button"
  onClick={[MockFunction]}
  style={
    Object {
      "display": "block",
    }
  }
/>
`;
